src/hooks/useOverflowDetection/useOverflowDetection.ts

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect, useRef } from 'react'
```

Comment:
Do we want to write this file `useOverflowDetection` or should we use an existing library?